src/sqlancer/cnosdb/CnosDBLoggableFactory.java

```diff
@@ -0,0 +1,59 @@
+package sqlancer.cnosdb;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import sqlancer.cnosdb.query.CnosDBOtherQuery;
+import sqlancer.cnosdb.query.CnosDBQueryAdapter;
+import sqlancer.common.log.Loggable;
+import sqlancer.common.log.LoggableFactory;
+import sqlancer.common.log.LoggedString;
+import sqlancer.common.query.ExpectedErrors;
+import sqlancer.common.query.Query;
+
+public class CnosDBLoggableFactory extends LoggableFactory {
```

Comment:
I think inheriting from https://github.com/sqlancer/sqlancer/blob/67751dc3d83b880cd8c7ed53416b8b157a2ba2ea/src/sqlancer/common/log/SQLLoggableFactory.java#L9 and overriding methods whose implementation differs would prevent some code duplication here.

Reply:
I noticed that this is still unaddressed. Is it not possible to inherit because of some reason related to not using JDBC?

Reply:
At present, our implementation does not use JDBC. At the beginning of development, JDBC support was not perfect, so we used the http interface, and we will optimize it later.